redis/src/main/scala/zio/redis/api/Keys.scala

```diff
@@ -13,72 +13,293 @@ import zio.{ Chunk, ZIO }
 trait Keys {
   import Keys.{ Keys => _, _ }
 
-  final def del(a: String, as: String*): ZIO[RedisExecutor, RedisError, Long] = Del.run((a, as.toList))
-
-  final def dump(a: String): ZIO[RedisExecutor, RedisError, Chunk[Byte]] = Dump.run(a)
-
-  final def exists(a: String, as: String*): ZIO[RedisExecutor, RedisError, Boolean] = Exists.run((a, as.toList))
-
-  final def expire(a: String, b: Duration): ZIO[RedisExecutor, RedisError, Boolean] = Expire.run((a, b))
-
-  final def expireAt(a: String, b: Instant): ZIO[RedisExecutor, RedisError, Boolean] = ExpireAt.run((a, b))
-
-  final def keys(a: String): ZIO[RedisExecutor, RedisError, Chunk[String]] = Keys.Keys.run(a)
-
+  /**
+   * Removes the specified keys. A key is ignored if it does not exist.
+
+   *
+   * @param key one required key
+   * @param keys maybe rest of the keys
+   * @return The number of keys that were removed.
+   *
+   * @see [[unlink]]
+   * @see <a href="https://redis.io/commands/del">https://redis.io/commands/del</a>
+   */
+  final def del(key: String, keys: String*): ZIO[RedisExecutor, RedisError, Long] = Del.run((key, keys.toList))
+
+  /**
+   * Serialize the value stored at key in a Redis-specific format and return it to the user.
+   *
+   * @param key key
+   * @return bytes for value stored at key
+   *
+   * @see <a href="https://redis.io/commands/dump">https://redis.io/commands/dump</a>
+   */
+  final def dump(key: String): ZIO[RedisExecutor, RedisError, Chunk[Byte]] = Dump.run(key)
+
+  /**
+   * The number of keys existing among the ones specified as arguments. Keys mentioned multiple times and existing are
+   * counted multiple times.
+   *
+   * @param key one required key
+   * @param keys maybe rest of the keys
+   * @return The number of keys existing.
+   *
+   * @see <a href="https://redis.io/commands/exists">https://redis.io/commands/exists</a>
+   */
+  final def exists(key: String, keys: String*): ZIO[RedisExecutor, RedisError, Boolean] = Exists.run((key, keys.toList))
```

Comment:
Exists return type should be Int

Reply:
Well, it is according to the docs, but indicates whether or not the key exists, which IMHO should be encoded as boolean.

Reply:
So, when there are multiple keys, `exists` command can return an integer greater than 1, in that case our code would throw, failing to convert to Boolean. I'd think that we either change the type to Int or possibly drop varargs support.

      case object BoolOutput extends Output[Boolean] {

        override protected def tryDecode(respValue: RespValue): Boolean =
          respValue match {
            case RespValue.Integer(0) => false
            case RespValue.Integer(1) => true
            case other                => throw ProtocolError(s"$other isn't a boolean")
          }

      }

Reply:
Ahh, forgot about multikey. Please add this to the "tickets to open", and we'll fix it there. We can have a chunk of booleans instead.

Reply:
Ahh, forgot about multikey. Please add this to the "tickets to open", and we'll fix it there. We can have a chunk of booleans instead

Reply:
Picking up this thread again, if you could elaborate what you meant by chunk of booleans? I'd guess you mean a boolean for each key, but from the api we get a single integer for all.

Reply:
Ah, I see. Ignore the chunk part, just return integer.

Reply:
Sure, I've raised the review now.